### PR TITLE
Backport #65821 to 6.7

### PR DESCRIPTION
**Pull request by ciampo**
As per https://github.com/WordPress/gutenberg/pull/65821#issuecomment-2391118370

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,27 @@
 -   `ToolsPanel`: atomic one-step state update when (un)registering panels ([#65564](https://github.com/WordPress/gutenberg/pull/65564)).
 -   `Navigator`: fix `isInitial` logic ([#65527](https://github.com/WordPress/gutenberg/pull/65527)).
 -   `ToggleGroupControl`: Fix arrow key navigation in RTL ([#65735](https://github.com/WordPress/gutenberg/pull/65735)).
+-   `ToggleGroupControl`: indicator doesn't jump around when the layout around it changes ([#65175](https://github.com/WordPress/gutenberg/pull/65175)).
+-   `Composite`: fix legacy support for the store prop ([#65821](https://github.com/WordPress/gutenberg/pull/65821)).
```
Comment (mirka) [inline packages/components/CHANGELOG.md:14]: Not sure how we're supposed to deal with the changelog 😅 I assume we should just add the entry for the changes we're cherry-picking, but would that just cause more merge conflict hassles down the line?

Comment (ciampo) [reply packages/components/CHANGELOG.md:14]: > would that just cause more merge conflict hassles down the line?

I don't know, but it makes sense that this PR only adds the relevant CHANGELOG entry

**Comment by ciampo** [comment]
CI errors are caused by https://github.com/WordPress/gutenberg/pull/65839#issuecomment-2391297800 — I will rebase once the target branch has them solved.

Update: fix was added, I rebased to include it.

**Review by tyxla — APPROVED** [review]
👍

**Pull request merged**